ui/cryptomaterial/slide_action.go

```diff
@@ -0,0 +1,187 @@
+package cryptomaterial
+
+import (
+	"image"
+	"time"
+
+	"gioui.org/f32"
+	"gioui.org/gesture"
+	"gioui.org/io/pointer"
+	"gioui.org/layout"
+	"gioui.org/op"
+	"gioui.org/op/clip"
+)
+
+const defaultDuration = 300 * time.Millisecond
+
+type SwipeDirection int
+
+const (
+	SwipeLeft SwipeDirection = iota
+	SwipeRight
+)
+
+type Dragged func(dragDirection SwipeDirection)
+
+type SliceAction struct {
+	duration time.Duration
+	push     int
+	next     *op.Ops
+	nextCall op.CallOp
+	lastCall op.CallOp
+
+	t0     time.Time
```

Comment:
why is this called t0?

Reply:
t0 is a mathematical way of expressing a time variable at the beginning